hbw/app/services/hbw/task_notifier.rb

```diff
@@ -21,7 +21,13 @@ def call(widget, task_id, event_name, users = [])
           message[:cache_key] = cache_key
         end
 
+        need_to_check_assignee = event_name == 'assignment'
+
         users.each do |user|
+          if need_to_check_assignee
```

Comment:
`message[:is_assigned] = event_name == 'assignment' && user == assignee`
or move this check to separate method like `need_to_fetch_task?`.